aim/web/ui/src/modules/BaseExplorer/components/RangePanel/RangePanel.tsx

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import _ from 'lodash-es';
+
+import { QueryUIStateUnit } from 'modules/BaseExplorerCore/core-store';
+import { getQueryFromRanges } from 'modules/BaseExplorerCore/utils/getQueryFromRanges';
+import { getQueryStringFromSelect } from 'modules/BaseExplorerCore/utils/getQueryStringFromSelect';
+
+import { Button, Icon, Text } from 'components/kit';
+
+import { SequenceTypesEnum } from 'types/core/enums';
+
+import RangePanelItem from './RangePanelItem';
+import { IRangePanelProps } from './RangePanel.d';
+
+import './RangePanel.scss';
+
+function RangePanel(props: IRangePanelProps) {
+  const engine = props.engine;
+  const sequenceName: SequenceTypesEnum = engine.useStore(
+    engine.sequenceNameSelector,
+  );
+  const query: QueryUIStateUnit = engine.useStore(engine.queryUI.stateSelector);
+  const rangeState = engine.useStore(engine.ranges.stateSelector);
+  const isFetching: boolean =
+    engine.useStore(engine.pipelineStatusSelector) === 'fetching';
+
+  const onSubmit = React.useCallback(() => {
+    if (isFetching) {
+      //TODO: abort request
+      return;
+    } else {
+      engine.search({
+        q: getQueryStringFromSelect(query, sequenceName),
+        report_progress: false,
+        ...getQueryFromRanges(rangeState),
+      });
+      engine.ranges.methods.update({
+        ...rangeState,
+        isApplyButtonDisabled: true,
+      });
+    }
+  }, [engine, isFetching, query, sequenceName, rangeState]);
+
+  const stepItemConfig = React.useMemo(
+    () => ({
+      sliderTitle: 'Steps',
+      countInputTitle: 'Steps count',
+      countTitleTooltip: 'Number of steps to display',
+      sliderTitleTooltip:
+        'Training step. Increments every time track() is called',
+      type: 'step',
+    }),
+    [],
+  );
+
+  const indexItemConfig = React.useMemo(
+    () => ({
+      sliderTitle: 'Indices',
+      countInputTitle: 'Indices count',
+      countTitleTooltip: `Number of ${sequenceName} per step`,
+      sliderTitleTooltip: `Index in the list of ${sequenceName} passed to track() call`,
+      type: 'index',
+    }),
+    [sequenceName],
+  );
+
+  React.useEffect(() => {
+    const ranges: {
+      record?: { slice: [number, number]; density: number };
+      index?: { slice: [number, number]; density: number };
+    } = {};
+    if (props.rangesData?.ranges?.record_range_total) {
+      const { record_range_used, record_range_total } =
+        props.rangesData?.ranges;
+      const slice: [number, number] = [
```

Comment:
Please leave comments, its hard to  guess whats going on.

We have created a middleware in adapter layer to do those checks to make compatible python list ranges with js ranges, didnt understand why we need to.
Please give an example to understand clearly.